src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -234,6 +240,19 @@ def _load_settings():
         # Append the instance to the internal map.
         RgwClient._user_instances[RgwClient._SYSTEM_USERID] = instance
 
+    def __get_daemon_zone_info(self):  # type: () -> Dict[str, Any]
```

Comment:
Is it necessary to use name mangling here and for ``__get_daemon_zonegroup_map``? Maybe marking it as private with ``_get_daemon_zone_info`` is enough.

Reply:
My intention here was to make it clear that the method is private (at least in the same sense as other languages). What I wanted to avoid is the "protected" meaning because it makes no sense when there is no inheritance here and it wasn't my original intention.
With that said, I'm OK with renaming it as long as we agree that "_" is "python-private" XD.
This is what [PEP8](https://www.python.org/dev/peps/pep-0008/#id47) says about it:
"Generally, double leading underscores should be used only to avoid name conflicts with attributes in classes designed to be subclassed.

Note: there is some controversy about the use of __names (see below)."

Reply:
I'm fine with both, only wanted to clarify if there is a strong reason to do so.

@sebastian-philipp What do you think about this?

Reply:
There was only one place in the Ceph repo, where I used a double-underscore: It was to avoid a name conflict in a Mixin for the Orchestrator client.

One underscore should be enough here.

```suggestion
    def _get_daemon_zone_info(self):  # type: () -> Dict[str, Any]
```

Reply:
@votdev @sebastian-philipp OK. Done.

Reply:
> Is it necessary to use name mangling here and for `__get_daemon_zonegroup_map`? Maybe marking it as private with `_get_daemon_zone_info` is enough.

Just for clarification: `_get_daemon_zone_info` is not private.

 `__method_name` is _private_ and `_method_name` is _protected_. And in addition to that, `protected` for Python is not as strict as you might think. Calling a protected method this way still works: `object._method_name()`.

Reply:
> @votdev
> 
> > Is it necessary to use name mangling here and for `__get_daemon_zonegroup_map`? Maybe marking it as private with `_get_daemon_zone_info` is enough.
> 
> Just for clarification: `_get_daemon_zone_info` is not private.
> 
> `__method_name` is _private_ and `_method_name` is _protected_. And in addition to that, `protected` for Python is not as strict as you might think. Calling a protected method this way still works: `object._method_name()`.
> 
> ```python
> In [1]: class A(object): 
>    ...:     def _test(self): 
>    ...:         print('foobar') 
>    ...:                                                                                                                                                                                        
> 
> In [2]: a = A()                                                                                                                                                                                
> 
> In [3]: a._test()                                                                                                                                                                              
> foobar
> ```

Thanks, this is all known, the phrasing was chosen because it is the easiest way to say what i meant.

Reply:
Actually `__method_name` is not private, but mangled. There are no private methods in Python